phoenix-core/src/main/java/org/apache/phoenix/coprocessor/UngroupedAggregateRegionScanner.java

```diff
@@ -479,6 +486,10 @@ void deleteCForQ(Tuple result, List<Cell> results, UngroupedAggregateRegionObser
             delete.addColumns(deleteCF,  deleteCQ, ts);
             // force tephra to ignore this deletes
             delete.setAttribute(PhoenixTransactionContext.TX_ROLLBACK_ATTRIBUTE_KEY, new byte[0]);
+            // TODO: We need to set SOURCE_OPERATION_ATTRIB here also. The control will come here if
```

Comment:
> Good question, @shahrs87 . I think we should set the attribute there.

@gjacoby126  I started doing changes for setting source operation attribute here. When we drop a column, we create delete markers with delete type "DELETE_COLUMN" for the base table and also we delete metadata for that column in SYSCAT table. Is it reasonable to have the feature first for just deleting rows from data table and then we can extend it for drop columns use case. Please advise. Thank you !

Reply:
The drop column case can be a separate JIRA.